ui-lib/src/main/java/co/electriccoin/zcash/ui/common/repository/FlexaRepository.kt

```diff
@@ -0,0 +1,109 @@
+    private val scope = CoroutineScope(Dispatchers.IO + SupervisorJob())
+
+    private val publishableKey: String?
+        get() = BuildConfig.ZCASH_FLEXA_KEY.takeIf { it.isNotEmpty() }
+
+    override fun init() {
+        scope.launch {
+            val configuration = getFlexaClientConfiguration()
+            if (configuration != null) {
+                Flexa.init(configuration)
+                Twig.info { "Flexa initialized" }
+
+                balanceRepository.state.collect { state ->
+                    Flexa.updateAppAccounts(
+                        arrayListOf(
+                            createFlexaAccount(
+                                zecBalance = state.totalBalance.convertZatoshiToZec().toDouble()
+                            )
+                        )
+                    )
+
+                    Twig.info { "Flexa updated by ${state.totalBalance.convertZatoshiToZec().toDouble()}" }
+                }
+            }
+        }
+    }
+
+    /**
+     * @return an instance of [FlexaClientConfiguration] or null if no publishable key set up
+     */
+    private suspend fun getFlexaClientConfiguration() =
+        publishableKey?.let { publishableKey ->
+            FlexaClientConfiguration(
+                context = application,
+                publishableKey = publishableKey,
+                theme =
+                    FlexaTheme(
+                        useDynamicColorScheme = true,
+                    ),
+                appAccounts = arrayListOf(createFlexaAccount(DEFAULT_ZEC_BALANCE)),
+                webViewThemeConfig =
+                    "{\n" +
+                        "    \"android\": {\n" +
+                        "        \"light\": {\n" +
+                        "            \"backgroundColor\": \"#100e29\",\n" +
+                        "            \"sortTextColor\": \"#ed7f60\",\n" +
+                        "            \"titleColor\": \"#ffffff\",\n" +
+                        "            \"cardColor\": \"#2a254e\",\n" +
+                        "            \"borderRadius\": \"15px\",\n" +
+                        "            \"textColor\": \"#ffffff\"\n" +
+                        "        },\n" +
+                        "        \"dark\": {\n" +
+                        "            \"backgroundColor\": \"#100e29\",\n" +
+                        "            \"sortTextColor\": \"#ed7f60\",\n" +
+                        "            \"titleColor\": \"#ffffff\",\n" +
+                        "            \"cardColor\": \"#2a254e\",\n" +
+                        "            \"borderRadius\": \"15px\",\n" +
+                        "            \"textColor\": \"#ffffff\"\n" +
+                        "        }\n" +
+                        "    }\n" +
+                        "}"
+            )
+        }
+
+    private suspend fun createFlexaAccount(zecBalance: Double) =
+        AppAccount(
+            accountId = getFlexaAccountId(),
+            displayName = "My Wallet",
```

Comment:
Is this our preferred `displayName` value for each user?

Reply:
I dont know yet :D

Reply:
done